demo/sorting.html

```diff
@@ -39,35 +39,36 @@
   </virtual-scroller>
 
   <script type="module">
-    import '../virtual-scroller-element.js';
+    import { ItemSource } from '../virtual-scroller-element.js';
     import './contacts/contact-element.js';
 
     const virtualScroller = document.querySelector('virtual-scroller');
 
     const onMoveup = (event) => moveContact(event.target.contact, -1);
     const onMovedown = (event) => moveContact(event.target.contact, 1);
-    virtualScroller.updateElement = (element, index) => {
+    virtualScroller.updateElement = (element, contact) => {
       element.addEventListener('moveup', onMoveup);
       element.addEventListener('movedown', onMovedown);
-      element.contact = myContacts[index];
+      element.contact = contact;
     };
 
     let myContacts = null;
     fetch('contacts/contacts.json').then((resp) => resp.json()).then((contacts) => {
       myContacts = contacts;
-      virtualScroller.totalItems = myContacts.length;
+      toggleElementKey();
     });
 
     function moveContact(contact, offset) {
       const i = myContacts.indexOf(contact);
       const newI = Math.max(0, Math.min(myContacts.length, i + offset));
       myContacts.splice(i, 1); // remove it
       myContacts.splice(newI, 0, contact); // add it to new position
-      virtualScroller.requestReset(); // notify virtual-scroller
+      virtualScroller.itemsChanged(); // notify virtual-scroller
     }
 
     window.toggleElementKey = function toggleElementKey(useGuid) {
-      virtualScroller.elementKey = useGuid ? (index) => myContacts[index].guid : null;
+      const key = useGuid ? (contact) => contact.guid : (_, index) => index;
```

Comment:
Would contact work, or do we need contact.guid?

It feels like this demo is less interesting with the new architecture; to demo the bad version you actually have to explicitly make it have a bad key function. Not if that's worth fixing though.